CMakeLists.txt

```diff
@@ -591,7 +591,7 @@ set(ZEPHYR_CURRENT_MODULE_DIR)
 set(ZEPHYR_CURRENT_CMAKE_DIR)
 
 get_property(LIBC_LINK_LIBRARIES TARGET zephyr_interface PROPERTY LIBC_LINK_LIBRARIES)
-target_link_libraries(zephyr_interface INTERFACE ${LIBC_LINK_LIBRARIES})
+zephyr_link_libraries(${LIBC_LINK_LIBRARIES})
```

Comment:
Yeah, this (as you probably are aware) was just a hack to avoid inserting these ahead of the loop.